### PR TITLE
SA-2000: Add preset reports to the all reports modal

### PR DESCRIPTION
**Pull request by kchu1337**
### What Changed

- Added preset report to all reports modal
- Added creator field to preset reports
- Added tests

### How To Test

- Go to /signals/analytics?
- Check that the saved reports dropdown still has preset reports with Default creator tag.
- Click `View All Reports` button
- Click the `All Reports` tab.
- Scroll down.
- Verify that the preset reports are present and you can pin/create scheduled reports from them.

- Also check that when adding/deleting a saved report, the subscription endpoint gets called and usage is updated.

### To Do

- [ ] Any outstanding tasks go here
  > Manage reviewers' expectations on missing features, tests, bugs, late design changes, ...)

### CODE REVIEW

File: src/pages/reportBuilder/components/tests/ReportsListModal.test.js [hunk 1]
```diff
@@ -99,4 +99,16 @@ describe('ReportsListModal', () => {
       expect.objectContaining({ id: 0, name: 'My Saved Report' }),
     );
   });
+
+  it('Shows the preset reports in the All Reports tab', () => {
```
Comment (nstanard) [inline src/pages/reportBuilder/components/tests/ReportsListModal.test.js:103]: Tests pass but I'm noticing some warnings and errors in the output.

```
console.warn
    Matchbox: Deprecated prop "flat" of "Button"
    Use `variant` instead

      at D (node_modules/@sparkpost/matchbox-hibana/matchbox.js:1:2489)
      at Object.r [as flat] (node_modules/@sparkpost/matchbox-hibana/matchbox.js:1:2757)
      at checkPropTypes (node_modules/prop-types/checkPropTypes.js:61:42)
      at validatePropTypes (node_modules/react/cjs/react.development.js:1714:7)
      at Object.createElementWithValidation [as createElement] (node_modules/react/cjs/react.development.js:1813:5)
      at id (node_modules/styled-components/src/models/Keyframes.js:20:51)
      at id (node_modules/styled-components/src/models/Keyframes.js:20:51)

  console.error
    Warning: React does not recognize the `isType` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `istype` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
        in a (created by LinkAnchor)
        in LinkAnchor (created by Context.Consumer)
        in Link (created by Styled(Text))
        in Styled(Text) (created by HibanaUnstyledLink)
        in HibanaUnstyledLink (at UnstyledLink.js:15)
        in UnstyledLink (at PageLink.js:16)
        in PageLink (created by Styled(UnstyledLink))
        in Styled(UnstyledLink) (created by ActionList.Action)
        in ActionList.Action (at ActionList.js:23)
        in ActionList.Action (at ReportsListModalTabs.js:78)
        in div (created by styled.div)
        in styled.div (created by ActionList)
        in ActionList (at ActionList.js:15)
        in ActionList (at ReportsListModalTabs.js:76)
        in div (created by Transition)
        in div (created by styled.div)
        in styled.div (created by Transition)
        in div (created by Box)
        in Box (created by Transition)
        in Transition (created by Popover.Content)
        in Popover.Content (created by PopoverOverlay)
        in div (created by Box)
        in Box (created by PopoverOverlay)
        in span (created by Box)
        in Box (created by PopoverOverlay)
        in PopoverOverlay (created by HibanaPopover)
        in HibanaPopover (at Popover.js:15)
        in Popover (at ReportsListModalTabs.js:65)
        in Actions (at ReportsListModalTabs.js:225)
        in td (created by styled.td)
        in styled.td (created by HibanaTableCell)
        in HibanaTableCell (at Table.js:67)
        in Table.Cell (created by Box)
        in Box (at Box.js:9)
        in Box (at ReportsListModalTabs.js:101)
        in tr (created by styled.tr)
        in styled.tr (created by HibanaTableRow)
        in HibanaTableRow (at Table.js:89)
        in Table.Row (at ReportsListModalTabs.js:241)
        in getAllReportRowComponent (at Collection.js:175)
        in tbody (at TableCollection.js:30)
        in TableBody (at Collection.js:166)
        in table (created by styled.table)
        in styled.table (created by HibanaTable)
        in div (created by Styled(Box))
        in Styled(Box) (created by HibanaTable)
        in HibanaTable (at Table.js:56)
        in Table (at Collection.js:164)
        in div (at Collection.js:192)
        in Collection (created by Context.Consumer)
        in withRouter(Collection) (at TableCollection.js:82)
        in TableCollection (at ReportsListModalTabs.js:245)
        in AllReportsTab (at ReportsListModal.js:40)
        in ModalContentContainer (at ReportsListModal.js:54)
        in div (created by Box)
        in Box (created by Modal.Content)
        in Modal.Content (at Modal.js:53)
        in Modal.Content (at ReportsListModal.js:53)
        in div (created by Box)
        in Box (created by Transition)
        in div (created by Styled(Box))
        in Styled(Box) (created by Transition)
        in Transition (created by Modal)
        in div (created by ForwardRef(FocusLockUI))
        in ForwardRef(FocusLockUI) (created by ForwardRef(FocusLockUICombination))
        in ForwardRef(FocusLockUICombination) (created by Styled(Component))
        in Styled(Component) (created by Modal)
        in div (created by styled.div)
        in styled.div (created by Modal)
        in div (created by Box)
        in Box (created by Modal)
        in div (created by styled.div)
        in styled.div (created by TouchScrollable)
        in TouchScrollable (created by Modal)
        in HibanaPortal (created by Modal)
        in Modal (at Modal.js:35)
        in Modal (at ReportsListModal.js:51)
        in ReportsListModal (at ReportsListModal.test.js:33)
        in Router (created by MemoryRouter)
        in MemoryRouter (at TestApp.js:20)
        in Poll (created by ConnectFunction)
        in ConnectFunction (at Providers.js:19)
        in ErrorBoundary (at Providers.js:18)
        in Unknown (created by HibanaThemeProvider)
        in MatchboxStyleSheetManager (created by HibanaThemeProvider)
        in HibanaThemeProvider (at ThemeProvider.js:9)
        in ThemeProvider (at Providers.js:17)
        in Provider (created by ConnectFunction)
        in ConnectFunction (at Providers.js:16)
        in ReactQueryCacheProvider (at QueryCache.js:15)
        in QueryCacheProvider (at Providers.js:15)
        in Provider (at Providers.js:14)
        in Providers (at TestApp.js:19)
        in TestApp (at ReportsListModal.test.js:32)

      at printWarning (node_modules/react-dom/cjs/react-dom.development.js:88:30)
      at error (node_modules/react-dom/cjs/react-dom.development.js:60:5)
      at validateProperty$1 (node_modules/react-dom/cjs/react-dom.development.js:5549:7)
      at warnUnknownProperties (node_modules/react-dom/cjs/react-dom.development.js:5595:21)
      at validateProperties$2 (node_modules/react-dom/cjs/react-dom.development.js:5619:3)
      at validatePropertiesInDevelopment (node_modules/react-dom/cjs/react-dom.development.js:5662:5)
      at setInitialProperties (node_modules/react-dom/cjs/react-dom.development.js:5941:5)

  console.error
    Warning: An update to HibanaPopover inside a test was not wrapped in act(...).

    When testing, code that causes React state updates should be wrapped into act(...):

    act(() => {
      /* fire events that update state */
    });
    /* assert on the output */

    This ensures that you're testing the behavior the user would see in the browser. Learn more at https://fb.me/react-wrap-tests-with-act
        in HibanaPopover (at Popover.js:15)
        in Popover (at ReportsListModalTabs.js:65)
        in Actions (at ReportsListModalTabs.js:225)
        in td (created by styled.td)
        in styled.td (created by HibanaTableCell)
        in HibanaTableCell (at Table.js:67)
        in Table.Cell (created by Box)
        in Box (at Box.js:9)
        in Box (at ReportsListModalTabs.js:101)
        in tr (created by styled.tr)
        in styled.tr (created by HibanaTableRow)
        in HibanaTableRow (at Table.js:89)
        in Table.Row (at ReportsListModalTabs.js:241)
        in getAllReportRowComponent (at Collection.js:175)
        in tbody (at TableCollection.js:30)
        in TableBody (at Collection.js:166)
        in table (created by styled.table)
        in styled.table (created by HibanaTable)
        in div (created by Styled(Box))
        in Styled(Box) (created by HibanaTable)
        in HibanaTable (at Table.js:56)
        in Table (at Collection.js:164)
        in div (at Collection.js:192)
        in Collection (created by Context.Consumer)
        in withRouter(Collection) (at TableCollection.js:82)
        in TableCollection (at ReportsListModalTabs.js:245)
        in AllReportsTab (at ReportsListModal.js:40)
        in ModalContentContainer (at ReportsListModal.js:54)
        in div (created by Box)
        in Box (created by Modal.Content)
        in Modal.Content (at Modal.js:53)
        in Modal.Content (at ReportsListModal.js:53)
        in div (created by Box)
        in Box (created by Transition)
        in div (created by Styled(Box))
        in Styled(Box) (created by Transition)
        in Transition (created by Modal)
        in div (created by ForwardRef(FocusLockUI))
        in ForwardRef(FocusLockUI) (created by ForwardRef(FocusLockUICombination))
        in ForwardRef(FocusLockUICombination) (created by Styled(Component))
        in Styled(Component) (created by Modal)
        in div (created by styled.div)
        in styled.div (created by Modal)
        in div (created by Box)
        in Box (created by Modal)
        in div (created by styled.div)
        in styled.div (created by TouchScrollable)
        in TouchScrollable (created by Modal)
        in HibanaPortal (created by Modal)
        in Modal (at Modal.js:35)
        in Modal (at ReportsListModal.js:51)
        in ReportsListModal (at ReportsListModal.test.js:33)
        in Router (created by MemoryRouter)
        in MemoryRouter (at TestApp.js:20)
        in Poll (created by ConnectFunction)
        in ConnectFunction (at Providers.js:19)
        in ErrorBoundary (at Providers.js:18)
        in Unknown (created by HibanaThemeProvider)
        in MatchboxStyleSheetManager (created by HibanaThemeProvider)
        in HibanaThemeProvider (at ThemeProvider.js:9)
        in ThemeProvider (at Providers.js:17)
        in Provider (created by ConnectFunction)
        in ConnectFunction (at Providers.js:16)
        in ReactQueryCacheProvider (at QueryCache.js:15)
        in QueryCacheProvider (at Providers.js:15)
        in Provider (at Providers.js:14)
        in Providers (at TestApp.js:19)
        in TestApp (at ReportsListModal.test.js:32)

      at printWarning (node_modules/react-dom/cjs/react-dom.development.js:88:30)
      at error (node_modules/react-dom/cjs/react-dom.development.js:60:5)
      at warnIfNotCurrentlyActingUpdatesInDEV (node_modules/react-dom/cjs/react-dom.development.js:23284:7)
      at dispatchAction (node_modules/react-dom/cjs/react-dom.development.js:15656:9)
      at h (node_modules/@sparkpost/matchbox-hibana/matchbox.js:1:136022)
      at node_modules/@sparkpost/matchbox-hibana/matchbox.js:1:136185
      at invokeEventListeners (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:193:27)
```

**Comment by nstanard** [comment]
@kchu1337 Correct me if I'm wrong but - wouldn't we also want the Dashboard change report modal to show defaults as well?

image

image

File: src/pages/reportBuilder/ReportBuilder.js [hunk 2]
```diff
@@ -147,9 +147,11 @@ export function ReportBuilder({
   }, [refreshReportBuilder, reportOptions, isEmpty]);
 
   useEffect(() => {
-    getSubscription();
+    if (reportsStatus === 'success') {
```
Comment (kchu1337) [inline src/pages/reportBuilder/ReportBuilder.js:150]: When the API call starts, reports gets reset to [], this effect runs. When it finishes and updates state, this effect runs again. So I updated the condition so it only runs when the API finishes getting reports.

**Review by nstanard — APPROVED** [review]
Tested dashboard and analytics page. Changes reports on both pages. Changed to default and custom. LGTM!

**Pull request merged**